### ISSUE TITLE
unused local variables in anonymous functions

### ISSUE THREAD

**Issue by SailorMax** [phpmd/phpmd#500]
Can you add check unused local variables in anonymous functions?

Sample:
```php
$funcGetName = function($uid) use ($record_row)
{
    global $USERS;
};
```

Expected:
```
Avoid unused parameters such as '$uid'.
Avoid unused local variables such as '$record_row'.
Avoid unused local variables such as '$USERS'.
```

Actual: no any warnings.

thank you.

**Comment by ravage84** [comment]
Confirmed.

**Comment by bkdotcom** [comment]
Not sure if this should be a separate bug, but it's related to the use statement:
$funcGetName = function ($uid) use ($index) {
    if ($index == 2) {
       //
    }
};

// Avoid unused local variables such as $index

**Comment by Pablo-Lilao** [comment]
Probably the same bug for variables with short names.

Sample:

`function ()
{
    $b = 1;
};`

Expected:
`Avoid variables with short names like $b. Configured minimum length is 3.
Avoid unused local variables such as '$b'.
`

**Comment by danepowell** [comment]
Doesn't this rule already exist? And in fact it causes problems with closures: https://github.com/phpmd/phpmd/issues/278

**Comment by bkdotcom** [comment]
> [ozgurg](/ozgurg)

Probably related to the syntax error?

`new class`  ??

**Comment by ozgurg** [comment]
> 



> > [ozgurg](/ozgurg)
> 
> Probably related to the syntax error?
> 
> `new class` ??

No, it's an anonymous class. https://www.php.net/manual/en/language.oop5.anonymous.php

**Comment by bkdotcom** [comment]
> [ozgurg](/ozgurg)

D'oh...    I'll see myself out.
